enigma-cli/src/main/java/cuchaz/enigma/command/Command.java

```diff
@@ -41,22 +43,36 @@ protected static EnigmaProject openProject(Path fileJarIn, Path fileMappings) th
 			System.out.println("Reading mappings...");
 
 			MappingSaveParameters saveParameters = enigma.getProfile().getMappingSaveParameters();
-			EntryTree<EntryMapping> mappings = chooseEnigmaFormat(fileMappings).read(fileMappings, progress, saveParameters);
+			EntryTree<EntryMapping> mappings = readMappings(fileMappings, progress, saveParameters);
 
 			project.setMappings(mappings);
 		}
 
 		return project;
 	}
 
-	protected static MappingFormat chooseEnigmaFormat(Path path) {
-		if (Files.isDirectory(path)) {
-			return MappingFormat.ENIGMA_DIRECTORY;
-		} else if ("zip".equalsIgnoreCase(MoreFiles.getFileExtension(path))) {
-			return MappingFormat.ENIGMA_ZIP;
+	protected static EntryTree<EntryMapping> readMappings(Path path, ProgressListener progress, MappingSaveParameters saveParameters) throws Exception {
+		List<Exception> suppressed = new ArrayList<>();
+
+		if ("zip".equalsIgnoreCase(MoreFiles.getFileExtension(path))) {
+			return MappingFormat.ENIGMA_ZIP.read(path, progress, saveParameters);
 		} else {
-			return MappingFormat.ENIGMA_FILE;
+			for (MappingFormat format : MappingFormat.getReadableFormats()) {
+				try {
+					return format.read(path, progress, saveParameters);
+				} catch (Exception e) {
+					suppressed.add(e);
+				}
```

Comment:
This is awful, it should detect the format before trying to read it like this.

Mapping-io has a thing to detect mapping formats, either use something similar or come back to this change once moved over to mapping-io. 

For now I see little benefit in this PR.

Reply:
It's definitely a bit hacky, but since we'll be switching to mapping-io soon anyway, I don't see a point in investing the time to write a proper mapping format detector.

> For now I see little benefit in this PR
Users can check other mapping formats for correctness ;)